packages/library/src/commonMain/kotlin/io/realm/internal/BaseRealmImpl.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.internal
+
+import io.realm.BaseRealm
+import io.realm.Callback
+import io.realm.Cancellable
+import io.realm.RealmConfiguration
+import io.realm.RealmObject
+import io.realm.RealmResults
+import io.realm.VersionId
+import io.realm.interop.NativePointer
+import io.realm.interop.RealmInterop
+import kotlinx.coroutines.flow.Flow
+import kotlin.reflect.KClass
+
+@Suppress("UnnecessaryAbstractClass")
+public abstract class BaseRealmImpl internal constructor(
```

Comment:
Any reason this is public?

Reply:
This PR is targeting an integration branch, but cannot track the other tasks in a PR until the branch differs from master. 

I started to internalize stuff as part of #383. Rest will be handled as part of #56 and #22.  I have added it in the TODOs of this PR and will merge the spill overs into the TODOs of the integration PR.